internal/wazeroir/compiler.go

```diff
@@ -794,29 +790,33 @@ operatorSwitch:
 		defaultTargetFrame := c.controlFrames.get(int(l))
 		defaultTargetFrame.ensureContinuation()
 		defaultTargetDrop := c.getFrameDropRange(defaultTargetFrame, false)
-		defaultTarget := defaultTargetFrame.asLabel()
-		c.result.LabelCallers[defaultTarget.ID()]++
+		defaultTargetID := defaultTargetFrame.asLabelID()
+		c.result.LabelCallers[defaultTargetID]++
 
 		c.emit(
-			OperationBrTable{
-				Targets: targets,
-				Default: &BranchTargetDrop{
-					ToDrop: defaultTargetDrop, Target: defaultTarget,
-				},
-			},
+			NewOperationBrTable(
+				append([]uint64{uint64(defaultTargetID)}, targetLabels...),
+				append([]*InclusiveRange{defaultTargetDrop}, targetDrops...),
```

Comment:
see the comment on NewOperationBrTable. you should be able to avoid allocation by not using append here.